lib/scoped_search/query_builder.rb

```diff
@@ -485,6 +489,16 @@ def to_sql(builder, definition, &block)
             raise ScopedSearch::QueryNotSupported, "Don't know how to handle this operator node: #{operator.inspect} with #{children.inspect}!"
           end
         end
+
+        private
+
+        def validate_value(field, value)
+          validator = field.validator
+          if field.validator && field.only_explicit
+            valid = validator.call(value)
+            raise ScopedSearch::QueryNotSupported, "Value '#{field.field}' is not valid for field '#{value}'" unless valid
```

Comment:
The field name and value variables are the wrong way around here - it should say that the value is not valid for the field.